modules/openapi-generator/src/main/resources/JavaSpring/nullableAnnotation.mustache

```diff
@@ -0,0 +1 @@
+{{^required}}{{^useOptional}}{{#openApiNullable}}{{^isNullable}}@Nullable {{/isNullable}}{{/openApiNullable}}{{^openApiNullable}}@Nullable {{/openApiNullable}}{{/useOptional}}{{/required}}
```

Comment:
I think for JsonNullable and when you using using Collection which could no be null it make also no sense to set this annotation

Reply:
And one more question do you think this make sense when you using bean validation?

Reply:
Hi @MelleD,

Thank for your quick review, appreciate it.

> for JsonNullable

Indeed, that was already taken into consideration. If the field is either wrapped with `JsonNullable` or `Optional` it is not annotated with `@Nullable`. Please, see the tests at the 1st commit.

> using Collection

Actually, it is not about just collection but any property with default value. For collections it was more obvious. My fault, thank you for catching it. Fixed that at the 4th commit (and updated samples again at the 5th commit)

> when you using bean validation

If I got you right, it is not intersected. See: any property which is not required might be `null`. Any validation (except `@NotNull` of course) bypasses and should bypass any `nulls`, which is correct. So, the bean validation doesn't prevent `NPEs` and it is worth to annotate such fields, I guess

Reply:
Hi @MelleD,

Thank for your quick review, appreciate it.

for JsonNullable

Indeed, that was already taken into consideration. If the field is either wrapped with JsonNullable or Optional it is not annotated with @Nullable. Please, see the tests at the 1st commit.

using Collection

Actually, it is not about just collection but any property with default value. For collections it was more obvious. My fault, thank you for catching it. Fixed that at the 4th commit (and updated samples again at the 5th commit)

when you using bean validation

If I got you right, it is not intersected. See: any property which is not required might be null. Any validation (except @NotNull of course) bypasses and should bypass any nulls, which is correct. So, the bean validation doesn't prevent NPEs and it is worth to annotate such fields, I guess

Reply:
> Indeed, that was already taken into consideration. If the field is either wrapped with `JsonNullable` or `Optional` it is not annotated with `@Nullable`. Please, see the tests at the 1st commit.

Ah I missed that, I just saw useOptional and thought maybe JsonNullable make also no sense :-)

> If I got you right, it is not intersected. See: any property which is not required might be null. Any validation (except @NotNull of course) bypasses and should bypass any nulls, which is correct. So, the bean validation doesn't prevent NPEs and it is worth to annotate such fields, I guess

Ok sounds reasonable. For a while I don't use nulls (I use Optional/Enum instead).

Reply:
I have some thoughts regarding `Optional` but let's discuss them as a separate story. For the moment my colleagues use just regular objects, and I would like to make their life easier with handling `NPEs`

Reply:
> I have some thoughts regarding Optional

It's fine there is no right or wrong everybody have own opinion about Optional.(https://medium.com/97-things/using-optional-nowhere-somewhere-or-everywhere-b1eb5645eab5)

I just say that I have no much experience with null handling ;)